build.gradle.kts

```diff
@@ -47,4 +60,72 @@ val integrationTestImplementation: Configuration by configurations.getting {
 }
 val integrationTestRuntimeOnly: Configuration by configurations.getting {
     extendsFrom(configurations.testRuntimeOnly.get())
-}
\ No newline at end of file
+}
+val containerName = findProperty("containerName") ?: "hivemq-k8s-test"
+val containerTag = findProperty("containerTag") ?: "snapshot"
+
+val producerK8sDockerImage: Configuration by configurations.creating {
+    isCanBeConsumed = false
+    isCanBeResolved = true
+    attributes {
+        attribute(Category.CATEGORY_ATTRIBUTE, objects.named("k8s-docker-image"))
+    }
+    extendsFrom(hivemq)
+}
+val producerDnsInitWaitDockerImage: Configuration by configurations.creating {
+    isCanBeConsumed = false
+    isCanBeResolved = true
+    attributes {
+        attribute(Category.CATEGORY_ATTRIBUTE, objects.named("dns-init-wait-docker-image"))
+    }
+    extendsFrom(operator)
+}
+val producerOperatorDockerImage: Configuration by configurations.creating {
+    isCanBeConsumed = false
+    isCanBeResolved = true
+    attributes {
+        attribute(Category.CATEGORY_ATTRIBUTE, objects.named("operator-docker-image"))
+    }
+    extendsFrom(operator)
+}
+
+
+val createRootlessK8sImageContext by tasks.registering(Sync::class) {
+    group = "container"
+    description = "Prepare container base image context"
+    into(layout.buildDirectory.dir("container/context"))
+    from("container")
+}
+
+val buildRootlessK8sImage by tasks.registering(Exec::class){
+    group = "container"
+    description = "Build docker image"
+    inputs.property("dockerImageName", containerName)
+    inputs.dir(createRootlessK8sImageContext.map { it.destinationDir })
+    workingDir(createRootlessK8sImageContext.map { it.destinationDir })
+    commandLine("docker","build","-f","broker.dockerfile","-t","${containerName}-rootless:${containerTag}",".")
+}
+
+val saveRootlessK8sImage by tasks.registering(Exec::class){
```

Comment:
As you have 2 tasks for building and saving the docker image, the tasks can never be up-to-date as inputs and outputs are needed. This means the tasks are always re-executed, which will always take a lot of time (especially for the `saveRootlessK8sImage` task).

Reply:
I leave it for next fix as will include changes to the images